services/QuillLMS/app/models/district.rb

```diff
@@ -4,14 +4,35 @@
 #
 # Table name: districts
 #
-#  id         :integer          not null, primary key
-#  name       :string
-#  token      :string
-#  created_at :datetime
-#  updated_at :datetime
-#  clever_id  :string
+#  id             :integer          not null, primary key
+#  city           :string
+#  grade_range    :string
+#  name           :string
+#  phone          :string
+#  state          :string
+#  token          :string
+#  total_schools  :integer
+#  total_students :integer
+#  zipcode        :string
+#  created_at     :datetime
+#  updated_at     :datetime
+#  clever_id      :string
+#  nces_id        :integer
 #
 class District < ApplicationRecord
 
-  has_and_belongs_to_many :users
+  has_many :schools
+  has_many :district_admins, class_name: 'DistrictAdmin', dependent: :destroy
+  has_many :admins, through: :district_admins, source: :user
+
+  scope :by_name, ->(name) { where('lower(name) LIKE ?', "%#{name.downcase}%") }
+  scope :by_city, ->(city) { where('lower(city) LIKE ?', "%#{city.downcase}%") }
+  scope :by_state, ->(state) { where(:state => state.upcase) }
+  scope :by_zipcode, ->(zipcode) { where(:zipcode => zipcode) }
+  scope :by_nces_id, ->(nces_id) { where(:nces_id => nces_id) }
+
+  def total_invoice
+    schools.sum { |s| s&.subscription&.payment_amount || 0 } / 100
```

Comment:
Another integer division item to watch out for. Make the denominator a float to avoid.
```suggestion
    schools.sum { |s| s&.subscription&.payment_amount || 0 } / 100.0
```